context.go

```diff
@@ -30,7 +37,7 @@ type Context struct {
 }
 
 // NewContext returns a Context associated with root.
-func NewContext(root string) (*Context, error) {
+func NewPathContext(root string) (Context, error) {
```

Comment:
I don't like this change. This will always be `NewContext`. I'll just merge and fix, but your code will break.